toolchain/check/convert.cpp

```diff
@@ -392,6 +392,12 @@ static auto ConvertStructToStructOrClass(Context& context,
   auto& sem_ir = context.sem_ir();
   auto src_elem_fields = sem_ir.inst_blocks().Get(src_type.fields_id);
   auto dest_elem_fields = sem_ir.inst_blocks().Get(dest_type.fields_id);
+  bool dest_has_vptr =
+      !dest_elem_fields.empty() &&
+      sem_ir.insts()
+              .GetAs<SemIR::StructTypeField>(dest_elem_fields.front())
+              .name_id == SemIR::NameId::Vptr;
```

Comment:
This might be worth factoring out into a helper function. Let's keep an eye on this and see if we want to do it in more places.